### PR TITLE
Vanishika Harlotikus

### PR DESCRIPTION
**Pull request by NPC1314**
## About The Pull Request

Court mage overhaul
* Old age wizards gets couple top end spells as reward for their life of study that younger ones doesn't
* All get some drip (hats, dark robes for old veteran mages unless already got the blue one)
* Wizard hat now helps them focus and they get anxious when forgetting where they put it in their dementia
(2 int stored in their hat, gets stressed when not wearing it. Encourages wearing it instead of helmet but doesnt force you to)
* Having a long beard gives a slight bonus (tied in to the lore and because its funny)
* Most spells gets more bling, sorted visuals, sounds, icons to make stuff cooler and more evident whats going on 
( like some sparks for lightning bolt hitting, door opening sound & spell icon for Knock, etc)
* Dwarves can now be court mages. Allowing elven rulers were a terrible mistake, this is the only way to balance it.
* Lorebook added re : magic giving some flavor to this stuff
* Role desc updated to the new standard
* Sprites added for several spells, fudged sideview of hat sorted
* Wizard now spawns with scrying orb and ring of protection, but loses a point in polearm. Already powerful as heck and
should always use magic to solve stuff if able.
* Middle aged and Old only
* PQ 0 required, if you can't maintain that you cant be trusted with knock and fireballs
* Shaving sounds added and powergamers who go beard and try to shave it get their just desserts.

Also fixed minor thing with the new lorebooks, having a separate subtype was pointless clutter, a var served the same purpose.

## Why It's Good For The Game

Big chested youthful court mages with nymphomania is the cursed 19 year old veteran ranger of our time. I will not stand for it unchallenged, other styles of mages must get some encouragement!

Also needed to be a little more spicy, was too bland.

People can still play their cringe nitemaiden mages but for once, the cool ones gets a carrot.
courtmage

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by Plaguewalker** [comment]
Might I suggest making court mage middle age and old only?

**Comment by NPC1314** [comment]
Considered that, and you know what, yes. You can still play young apprentice wandering mage and go apply for the job if you want the fresh faced neophyte thing

**Comment by Cre77** [comment]
so basically playing female makes you unable to get forcewall?

**Comment by Optimism333** [comment]
I think most of the changes to buff an older mage are good, as well as encouraging them to wear their hat, but I agree that it's not good practice to gate an additional spell behind having a beard.

**Comment by NPC1314** [comment]
Nah I like the beard thing. Let the people who want to be sweaty little goblins shave and see what happens.

**Comment by CormanQuewye** [comment]
I love the idea of tying some stuff to items, but I'd make swap it from the hat to the staff (the whole concept of a magical focus). The beard thing is also just unecessarily and 'because it's funny' hardly sounds a like a lore or roleplay sensible reason.

The age thing is hit-or-miss with me. With the current court wizard desc in the selection screen it feels a lot like someone at a table saying 'but I don't like your character concept'. If Stonekeep's wizard lore gets swapped to something like there's an actual wizardy guild/college situation going on and courts around the world get their wizards from there, then the middle-age/old thing starts making a lot more sense.

File: code/modules/jobs/job_types/roguetown/nobility/magician.dm [hunk 1]
```diff
@@ -58,4 +72,47 @@
...
+		C.add_stress(/datum/stressevent/wheresmyhat)
+
+/datum/stressevent/wheresmyhat
+	timer = 0
+	stressadd = 1
+	desc = "<span class='red'>I miss my hat....</span>"
+
+/atom/movable/screen/alert/status_effect/buff/thinking_cap
+	name = "My mind is clear"
+	desc = "<span class='nicegreen'>My hat deflects mind-clouding rays of Ziso...</span>\n"
```
Comment (ThePainkiller) [inline code/modules/jobs/job_types/roguetown/nobility/magician.dm:116]:
```suggestion
	desc = "<span class='nicegreen'>My hat deflects mind-clouding rays of Zizo...</span>\n"
```

**Review by ThePainkiller — CHANGES_REQUESTED** [review]

**Comment by ThePainkiller** [comment]
Forcewall is a spell all mages should have. Make it a different spell that gets gated behind the beard, it wouldn't be fair for old hags.

**Comment by NPC1314** [comment]
> Forcewall is a spell all mages should have. Make it a different spell that gets gated behind the beard, it wouldn't be fair for old hags.

No court mages have it currently, or has ever had it, so its not possible for the old hags to feel the loss of something they never had. They going to be old elf anyways and doing the same old orgies, they clearly dont need the spell.

**Comment by Cre77** [comment]
> > Forcewall is a spell all mages should have. Make it a different spell that gets gated behind the beard, it wouldn't be fair for old hags.
> 
> No court mages have it currently, or has ever had it, so its not possible for the old hags to feel the loss of something they never had. They going to be old elf anyways and doing the same old orgies, they clearly dont need the spell.

regardless, this is some very outdated, targeted sexism. such behavior should be punished with actual pq dunks, not giving less functionality to people who just picked the wrong option in chargen. and even then - the people that play court magician to fulfil their niche fetishes aren't going to be stopped by that. it's not like they're going to be using those spells to begin with, seeing as they'll be locked in a bedroom for most of the round

**Comment by Plaguewalker** [comment]
> Forcewall is a spell all mages should have. Make it a different spell that gets gated behind the beard, it wouldn't be fair for old hags.

Maybe greater fireball for the beard?

**Comment by Optimism333** [comment]
> > Forcewall is a spell all mages should have. Make it a different spell that gets gated behind the beard, it wouldn't be fair for old hags.
> 
> No court mages have it currently, or has ever had it, so its not possible for the old hags to feel the loss of something they never had. They going to be old elf anyways and doing the same old orgies, they clearly dont need the spell.

It's not good practice to exclude a bonus spell from certain players just because you have little faith that someone can play an elder female mage character in a way that doesn't involve erp. I'm very much on board with the idea of rewarding older wizards with some bonuses, and a bit of thematic flair, but you should apply these bonuses evenly.

### In Summary ###
- Adding more reward for being an older mage who has studied/practiced more as a tradeoff for being more physically frail is good
- Gatekeeping a bonus spell behind having a beard (a male only option) is bad

**Comment by sanshoom** [comment]
would it be possible to randomly remove spells if the court mage had sex, cause they seem to do that way more often than being an actual wizard (plus having sex as a noc worshipper is a sin)

**Comment by Stutternov** [comment]
As some feedback;

I think linking the spell behind specific beard style is a bit odd, while I get it this encourages a specific playstyle my opinion would be making each gender have specific unique spells as then, at least, it's a different niche. I get the aesthetic encouragement, but I don't think this is particularly the way to do it.

Also, if the issue people have highlighted is 'sexualized old hags' etc as I saw discussed on discord even, give them the negative sex drive trait (I forget what it's called in code off-hand). These _are_ magicians that probably would vow to celibacy for knowledge and 'meditation' to master the arcane. (Aka redditors).

**Comment by ThePainkiller** [comment]
> would it be possible to randomly remove spells if the court mage had sex, cause they seem to do that way more often than being an actual wizard (plus having sex as a noc worshipper is a sin)

This would only be possible by having their virgin flag set to TRUE by default (it's currently FALSE for everyone by default), and then making a virginity loss check, but I'd have to see if there's even any procs to _remove_ or _"unlearn"_ spells so that it takes them away, as a punishment from Noc for engaging in carnal desire (lore wise makes sense)

**Comment by casualspacestation14enjoyer** [comment]
image
make court magician one of these, they now have an oath of chastity

**Comment by NPC1314** [comment]
Truly a nontroversy, some beards getting a second rate spell gift is not a balance problem, its not a gameplay problem, its not 
a moral problem, Lord give me strength.

What the PR does is to fix some issues and add some flavor and flair, thats obvious in the description, not truly solve the issue of horny maiden wizard. The intent is to incentivize characters that looks distinct and are less likely to engage in that behaviour, a soft nudge rather than a hard lock.

Its give and take,  find it amusing to give a carrot to beard-lovers and adding little low-impact quirks thats part of what motivates me to do any content at all, mirror design where fairness over percieved slights is not in any way better design and very uninspiring.

Sure, could be another spell, but I looked at the list of the ones court mage never had and forcewall feels like it fits the bill of useful but not very abusable, unlike greater fireball, knock or jaunt for example.

**Comment by NPC1314** [comment]
> > would it be possible to randomly remove spells if the court mage had sex, cause they seem to do that way more often than being an actual wizard (plus having sex as a noc worshipper is a sin)
> 
> This would only be possible by having their virgin flag set to TRUE by default (it's currently FALSE for everyone by default), and then making a virginity loss check, but I'd have to see if there's even any procs to _remove_ or _"unlearn"_ spells so that it takes them away, as a punishment from Noc for engaging in carnal desire (lore wise makes sense)

The proc for removing spells is used in this very PR
						if(user.mind && user.mind.assigned_role == "Court Magician")
							var/mob/living/carbon/V = src
							V.add_stress(/datum/stressevent/dwarfshaved)
							V.mind.RemoveSpell(/obj/effect/proc_holder/spell/targeted/forcewall)

**Comment by CormanQuewye** [comment]
> Truly a nontroversy, some beards getting a second rate spell gift is not a balance problem, its not a gameplay problem, its not a moral problem, Lord give me strength.
> 
> What the PR does is to fix some issues and add some flavor and flair, thats obvious in the description, not truly solve the issue of horny maiden wizard. The intent is to incentivize characters that looks distinct and are less likely to engage in that behaviour, a soft nudge rather than a hard lock.
> 
> Its give and take, find it amusing to give a carrot to beard-lovers and adding little low-impact quirks thats part of what motivates me to do any content at all, mirror design where fairness over percieved slights is not in any way better design and very uninspiring.
> 
> Sure, could be another spell, but I looked at the list of the ones court mage never had and forcewall feels like it fits the bill of useful but not very abusable, unlike greater fireball, knock or jaunt for example.

But I'm into playing bearded old bastards that spend their whole round in the baths.

**Review by Stutternov — APPROVED** [review]

**Comment by DAKKA-WAAAGH** [comment]
I won't demand for a reversion, but I would prefer if people are positively rewarded rather than punished in these cases. I agree wizards should not be wearing armor, but I do not agree that part of their important stat is tied into wearing a specific hat.

Perhaps return their Intelligence to normal and buff them slightly for wearing the hat, so that those who choose to wear "drip" aren't punished for doing so. For those that just want to be default wizard, they end up getting a cool minor buff and that's fine.

**Comment by Tipy1802** [comment]
I heavily disagree with this PR. People shouldn't be shoehorned in their customization options and forced to play only one generic trope of a character in a roleplaying game. Even more so when said trope only works for humans yet is forced on all races which further breaks immersion.

> Old age wizards gets couple top end spells as reward for their life of study that younger ones doesn't

I couldn't find what exactly this does in the code but I am very code illiterate so I cannot talk on the precise implementation but in general I don't really like the idea of soft forcing a mage to be old. Even more so when the role is called Court MAGICIAN and not WIZARD so they could be a sorcerer with a natural gift for magic. What makes this even worse is that this is applied to all races, even elves who even at a young age are probably older than most "old" humans

> Wizard hat now helps them focus and they get anxious when forgetting where they put it in their dementia
(2 int stored in their hat, gets stressed when not wearing it. Encourages wearing it instead of helmet but doesnt force you to)

This is just stupid. Forcing someone to wear one specific hat is just stupid. People shouldn't be punished for wanting to have different drip that fits their character more. If you wish to encourage mages to not be heavily armoured then they already are by virtue of not having training in armour. If that is not enough for you you can give them the exert doger trait. If even that is not enough they you could make a custom trait for them where if they are wearing non-light or any (if you want to be extreme) armour then they get some other buff/debuff. If your problem is with helmets specifically, then maybe there is a way to make the trait check for helmets specifically.

> Having a long beard gives a slight bonus (tied in to the lore and because its funny)

Utter nonsense that kills character diversity for no reason. Not everyone want to play Merlin, there should be more mage archetypes than just bearded old guy. It also doesnt make sense for elves who tend to not have much facial hair and in lore are naturally better at magic than the other races. Also I checked the code and only ONE elven beard works. I dont think forcing all elf mages to look exactly the same is a good idea.

> Wizard now spawns with scrying orb and ring of protection, but loses a point in polearm. Already powerful as heck and
should always use magic to solve stuff if able.

> Shaving sounds added and powergamers who go beard and try to shave it get their just desserts.

Lol, lmao even. Chosing to have or not to have a beard is a comsetic choice, not a powergame one. You are the one making it so having a beard is a powergame choice.

Oh and I disagree with dwarf court mage since dwarves are supposed to suck with magic and doing it in revange to elf "rulers" is petty and stupid. Besides I dont think an elf can be a king, only queen and thats because she is the wife of the king who may be married for diplomatic reasons. But whatever this isnt harmful, just not very thematic so I dont care as much.

In summery, this change forces mages to be only one stock trope, disregarding customisation, people's characters and the setting. I really do not see what benefits this brings and the harm is great. And yes giving disantvantages to specific cosmetic choices is forcing people to use them because this is a video game and people will gravitate to whats stronger. It just feels bad to get severely gimped for your cosmetic choices or wanting to rp a character outside of the ONE allowed archetype. Also these changes are not thoughtout fully because they make sense only for humans

You forgot to remove the scrying orb from the tower so the mage gets two now, I imagine that is not intentional

> Middle aged and Old only

Kinda makes sense for humans.... if ignore the posibility of one simply being gifted or a sorcerer. Makes no sense for elves, as I said before a young elf is still older than most humans and elves are just more naturally gifted when it comes to magic.

**Comment by RoboticVodka** [comment]
I meant to mention this before it even was merged but the stated purpose of:
"I will not stand for it unchallenged, other styles of mages must get some encouragement!"

Goes directly against what the PR does. It doesn't encourage different playstyles it encourages a single playstyle and that one playstyle only. (Old wizard with beard which also means only male old wizard.) Now having been a few days too some examples of how the PR's been working are visible.

If you're not old it puts you almost on par with drifter wizard in fact, especially if your hat is off. It's also been noted it's also practically a new player trap because someone may join as Court Wizard wanting a unique position and way to show it and they join as a middle aged wizard to be on par with the 3 other drifters wandering into town. Someone has already hired a drifter mage to replace their own court mage because they "Didn't even have greater fireball and jaunt." 

There should absolutely be a partial revert removing the old age requirement. I personally don't think anyone will cry over forcewall being behind a beard as it wasn't in Court Mage's base kit. The head wear requirement is a bit strange especially with how the head wear currently is like a dunce cap with stars on it. If https://github.com/Darkrp-community/OpenKeep/pull/307#issue-2389286672 is merged the Court Mage could also have the option of a circlet to satisfy their hat requirement I think that would be a wonderful compromise to offer more visual variety.

**Pull request merged**